src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs

```diff
@@ -206,7 +206,10 @@ private void WriteMethodBody(IMethodDefinition method)
                 Write("throw new ");
                 if (_forCompilationIncludeGlobalprefix)
                     Write("global::");
-                Write("System.PlatformNotSupportedException(); ");
+                if(_platformNotSupportedExceptionMessage == null)
+                    Write("System.PlatformNotSupportedException();");
+                else
+                    Write($"System.PlatformNotSupportedException(\"{_platformNotSupportedExceptionMessage}\"); ");
```

Comment:
@ericstj I can't get it working by just passing SR. For instance, I'm trying it for System.Security.Cryptography.OpenSsl and this is the line generated in System.Security.Cryptography.OpenSsl.notsupported.cs:
`public DSAOpenSsl() { throw new System.PlatformNotSupportedException(SR.Cryptography_InvalidDsaParameters_MissingFields); }`

This generates `error CS0103: The
 name 'SR' does not exist in the current context` Where is SR supposed to be defined in this project?